src/tools/clippy/clippy_lints/src/methods/mod.rs

```diff
@@ -2145,12 +2145,16 @@ impl<'tcx> LateLintPass<'tcx> for Methods {
                 // one of the associated types must be Self
                 for &(predicate, _span) in cx.tcx.explicit_item_bounds(def_id) {
                     if let ty::PredicateKind::Projection(projection_predicate) = predicate.kind().skip_binder() {
+                        let assoc_ty = match projection_predicate.term {
+                          ty::Term::Ty(ty) => ty,
+                          ty::Term::Const(c) => c.ty,
```

Comment:
Hmm... I wonder if we can test this... maybe

```rust
#![feature(adt_const_params)]

#[derive(PartialEq, Eq)]
struct Foo;
struct Bar<const C: Foo>;
impl Foo {
    fn new() -> Bar<{ Foo }> {
        Bar
    }
}
```

but that already doesn't lint on nightly clippy, so I'm guessing you can just skip constants with a `continue`